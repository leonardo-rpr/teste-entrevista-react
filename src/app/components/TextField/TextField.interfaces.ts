export interface TextFieldInterface {
  label: string;
  value: string;
  placeholder: string;
  onChange: (event: any) => void;
  error?: string | null;
  theme?: any;
}

export interface TextFieldStylesInterface {
  theme?: any;
}
