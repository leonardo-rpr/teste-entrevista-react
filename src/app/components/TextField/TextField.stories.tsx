import React from 'react';
import { withKnobs, text } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';
import TextField from './TextField';

storiesOf('Form', module)
  .addDecorator(withKnobs)
  .add('TextField', () => (
    <div style={{ padding: '20px' }}>
      <TextField
        label={text('Label', 'E-MAIL')}
        placeholder={text('Placeholder', 'leonardo@proenem.com')}
        value=""
        onChange={() => {}}
      />
    </div>
  ));
