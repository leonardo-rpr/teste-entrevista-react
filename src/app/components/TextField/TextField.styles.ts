import styled from 'styled-components';
import { TextFieldStylesInterface } from './TextField.interfaces';

export const Field = styled.div<TextFieldStylesInterface>`
  display: flex;
  flex-direction: column;
  margin-bottom: 20px;
`;

export const Label = styled.label<TextFieldStylesInterface>`
  font-size: 14px;
  margin-bottom: 10px;
  color: #f58843;
  text-transform: uppercase;
`;

export const Input = styled.input<TextFieldStylesInterface>`
  padding: 15px;
  border-radius: 4px;
  border: 1px solid #f58843;
  color: #f58843;
  outline: none;

  ::placeholder {
    color: #f58843;
    opacity: 0.4;
  }
`;
