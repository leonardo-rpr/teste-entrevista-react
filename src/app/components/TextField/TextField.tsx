import React from 'react';
import { Field, Label, Input, Error } from './TextField.styles';

import { TextFieldInterface } from './TextField.interfaces';

const TextField: React.FC<TextFieldInterface> = ({
  label,
  value,
  placeholder,
  onChange,
  error,
  touched,
  ...props
}) => (
  <Field>
    {label && <Label>{label}</Label>}
    <Input
      value={value}
      onChange={onChange}
      placeholder={placeholder}
      error={!!(touched && error)}
      {...props}
    />
    {!!(touched && error) && <Error>{error}</Error>}
  </Field>
);

TextField.defaultProps = {
  label: '',
};

export default TextField;
