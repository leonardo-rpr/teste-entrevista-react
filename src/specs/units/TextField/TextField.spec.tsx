import 'jest-styled-components';
import React from 'react';
import { render } from '@testing-library/react';
import TextField from '../../../app/components/TextField';

describe('TextField', () => {
  it('Should render correctly', () => {
    const placeholder = 'johndoe@proenem.com';
    const { container } = render(<TextField placeholder={placeholder} />);
    expect(container.firstChild.lastChild.getAttribute('placeholder')).toBe(placeholder);
    expect(container).toMatchSnapshot();
  });

  it('Should render correctly with label', () => {
    const label = 'Lorem Ipsum';
    const { container } = render(<TextField label={label} />);
    expect(container.firstChild.firstChild.textContent).toBe(label);
    expect(container).toMatchSnapshot();
  });

  it('Should render correctly with value', () => {
    const value = 'johndoe@proenem.com';
    const { container } = render(<TextField defaultValue={value} />);
    expect(container.firstChild.lastChild.getAttribute('value')).toBe(value);
    expect(container).toMatchSnapshot();
  });

  it('Should render correctly with other type', () => {
    const type = 'email';
    const { container } = render(<TextField type={type} />);
    expect(container.firstChild.lastChild.getAttribute('type')).toBe(type);
    expect(container).toMatchSnapshot();
  });
});
