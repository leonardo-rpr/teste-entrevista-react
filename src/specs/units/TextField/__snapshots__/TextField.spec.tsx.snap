// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField Should render correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-bottom: 20px;
}

.c1 {
  padding: 15px;
  border-radius: 4px;
  border: 1px solid #f2295b;
  color: #f2295b;
  outline: none;
}

.c1::-webkit-input-placeholder {
  color: #f2295b;
  opacity: 0.4;
}

.c1::-moz-placeholder {
  color: #f2295b;
  opacity: 0.4;
}

.c1:-ms-input-placeholder {
  color: #f2295b;
  opacity: 0.4;
}

.c1::placeholder {
  color: #f2295b;
  opacity: 0.4;
}

<div>
  <div
    class="c0"
  >
    
    <input
      class="c1"
      placeholder="johndoe@proenem.com"
      value=""
    />
  </div>
</div>
`;

exports[`TextField Should render correctly with label 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-bottom: 20px;
}

.c1 {
  font-size: 14px;
  margin-bottom: 10px;
  color: #f2295b;
  text-transform: uppercase;
}

.c2 {
  padding: 15px;
  border-radius: 4px;
  border: 1px solid #f2295b;
  color: #f2295b;
  outline: none;
}

.c2::-webkit-input-placeholder {
  color: #f2295b;
  opacity: 0.4;
}

.c2::-moz-placeholder {
  color: #f2295b;
  opacity: 0.4;
}

.c2:-ms-input-placeholder {
  color: #f2295b;
  opacity: 0.4;
}

.c2::placeholder {
  color: #f2295b;
  opacity: 0.4;
}

<div>
  <div
    class="c0"
  >
    <label
      class="c1"
    >
      Lorem Ipsum
    </label>
    <input
      class="c2"
      value=""
    />
  </div>
</div>
`;

exports[`TextField Should render correctly with other type 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-bottom: 20px;
}

.c1 {
  padding: 15px;
  border-radius: 4px;
  border: 1px solid #f2295b;
  color: #f2295b;
  outline: none;
}

.c1::-webkit-input-placeholder {
  color: #f2295b;
  opacity: 0.4;
}

.c1::-moz-placeholder {
  color: #f2295b;
  opacity: 0.4;
}

.c1:-ms-input-placeholder {
  color: #f2295b;
  opacity: 0.4;
}

.c1::placeholder {
  color: #f2295b;
  opacity: 0.4;
}

<div>
  <div
    class="c0"
  >
    
    <input
      class="c1"
      type="email"
      value=""
    />
  </div>
</div>
`;

exports[`TextField Should render correctly with value 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-bottom: 20px;
}

.c1 {
  padding: 15px;
  border-radius: 4px;
  border: 1px solid #f2295b;
  color: #f2295b;
  outline: none;
}

.c1::-webkit-input-placeholder {
  color: #f2295b;
  opacity: 0.4;
}

.c1::-moz-placeholder {
  color: #f2295b;
  opacity: 0.4;
}

.c1:-ms-input-placeholder {
  color: #f2295b;
  opacity: 0.4;
}

.c1::placeholder {
  color: #f2295b;
  opacity: 0.4;
}

<div>
  <div
    class="c0"
  >
    
    <input
      class="c1"
      value="johndoe@proenem.com"
    />
  </div>
</div>
`;
